cmd/echo.go

```diff
@@ -0,0 +1,56 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/jetstack/preflight/api"
+	"github.com/spf13/cobra"
+)
+
+var echoCmd = &cobra.Command{
+	Use:   "echo",
+	Short: "starts an echo server to test the agent",
+	Long: `The agent sends data to a server. This echo server
+can be used to act as the server part and echo the data received by the agent.`,
+	Run: echo,
+}
+
+var echoListen string
+
+func init() {
+	rootCmd.AddCommand(echoCmd)
+	echoCmd.PersistentFlags().StringVarP(
+		&echoListen,
+		"listen",
+		"l",
+		":8080",
+		"Address where to listen.",
+	)
+}
+
+func echo(cmd *cobra.Command, args []string) {
+	http.HandleFunc("/", echoHandler)
+	fmt.Println("Listening to requests at ", echoListen)
+	http.ListenAndServe(echoListen, nil)
+}
+
+func echoHandler(w http.ResponseWriter, r *http.Request) {
+	// if r.Method != http.MethodPost {
+	// 	http.Error(w, fmt.Sprintf("invalid method. Expected POST, received %s", r.Method), http.StatusBadRequest)
+	// 	return
+	// }
```

Comment:
Any reason to keep this around? (seems like the echo server shouldn't be concerned with validating the agent's behaviour?)

Reply:
That was old, now it is not commented but actual code.

The agent should send a POST, so it makes sense that the echo server returns an error if it receives something that is not a POST.

And in fact, this has been already useful. I was building the URL wrong in the agent (`http://localhost:8080//`, with double trailing `/`) and that was causing the http client to send a GET instead of a POST, even though I was using `http.Post()`. ¯\_(ツ)_/¯